pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1704,18 +1706,19 @@ func (cf *changeFrontier) maybeCheckpointJob(
 
 	// If the highwater has moved an empty checkpoint will be saved
 	//lint:ignore SA1019 deprecated usage
-	var checkpoint jobspb.ChangefeedProgress_Checkpoint
+	var legacyCheckpoint *jobspb.ChangefeedProgress_Checkpoint
+	var checkpoint *jobspb.TimestampSpansMap
 	if updateCheckpoint {
 		maxBytes := changefeedbase.SpanCheckpointMaxBytes.Get(&cf.FlowCtx.Cfg.Settings.SV)
-		checkpoint = cf.frontier.MakeCheckpoint(maxBytes, cf.sliMetrics.CheckpointMetrics)
+		legacyCheckpoint, checkpoint = cf.frontier.MakeCheckpoint(maxBytes, cf.sliMetrics.CheckpointMetrics, cf.evalCtx.Settings.Version)
```

Comment:
as discussed consider using the new checkpoint and having a method to convert it to the old representation.

Reply:
I made this change in a fixup commit, I'll squash it before merging.